akka-persistence-typed/src/main/scala/akka/persistence/typed/EventSourcedSignal.scala

```diff
@@ -52,13 +53,45 @@ final case class SnapshotFailed(metadata: SnapshotMetadata, failure: Throwable)
   def getSnapshotMetadata(): SnapshotMetadata = metadata
 }
 
+object SnapshotMetadata {
+
+  /**
+   * @param persistenceId id of persistent actor from which the snapshot was taken.
+   * @param sequenceNr sequence number at which the snapshot was taken.
+   * @param timestamp time at which the snapshot was saved, defaults to 0 when unknown.
+   *                  in milliseconds from the epoch of 1970-01-01T00:00:00Z.
+   */
+  def apply(persistenceId: String, sequenceNr: Long, timestamp: Long): SnapshotMetadata =
+    new SnapshotMetadata(persistenceId, sequenceNr, timestamp)
+
+  /**
+   * INTERNAL API
+   */
+  @InternalApi private[akka] def fromUntyped(metadata: akka.persistence.SnapshotMetadata): SnapshotMetadata =
```

Comment:
I guess `def fromUntyped` is ok being internal.

Reply:
I question relocating `SnapshotMetadata` to this signals file for design reasons, since the class itself is not a signal but a configuration of both `EventSourcedSignal`, `JournalProtocol.Message` and `SnapshotProtocol.Message`, it crosses all concerns.

Reply:
I'd consider
```scala
object SnapshotInteractions 
```
to align with its usage and we have `trait SnapshotInteractions` now.

Reply:
A case class can’t be changed (eg adding another field) in a binary compatible way. Therefor we try to stay away from the in public API apart from small trivial things that are unlikely to change.

Reply:
SnapshotMetadata is only exposed in two signals here in public api, and thats why I placed it in this file.

SnapshotInteractions is our internal stuff.